### PR TITLE
[front] - enh(builder): vault selection

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors the `VaultSelector` component to improve its functionality and user experience within the assistant builder. 

The main changes include:
- Updating the `VaultSelector` to use checkboxes instead of radio buttons, allowing for multiple vault selections.
- Implementing a dialog to inform users when changing source selection is not possible due to existing tool configurations.
- Enhancing the visual presentation of vault options, including better handling of disabled states.

**References:**
- https://github.com/dust-tt/tasks/issues/1361

## Risk

Low

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/assistant_builder/vaults/VaultSelector.tsx [hunk 1]
```diff
@@ -32,60 +31,75 @@ export function VaultSelector({
...
+                variant="checkable"
+                checked={isChecked ? "checked" : "unchecked"}
+                onChange={() => {
+                  if (!isDisabled) {
+                    setSelectedVault(isChecked ? "" : vault.sId);
+                  }
+                }}
+                disabled={isDisabled}
+              />
+              <div className="s-flex s-items-center s-gap-2">
```
Comment (flvndvd) [inline front/components/assistant_builder/vaults/VaultSelector.tsx:65]: No `s-`.

File: front/components/data_source/DeleteStaticDataSourceDialog.tsx [hunk 2]
```diff
@@ -59,7 +59,6 @@ export function DeleteStaticDataSourceDialog({
       title={`Removing ${name}`}
       onValidate={onDelete}
       isSaving={isLoading || isUsageLoading}
-      onCancel={onClose}
```
Comment (flvndvd) [inline front/components/data_source/DeleteStaticDataSourceDialog.tsx:62]: We might want to keep this one 😅.

**Review by flvndvd — APPROVED** [review]
LGTM, once comments are addressed.

File: front/components/data_source/DeleteStaticDataSourceDialog.tsx (hunk 2 above)
Comment (JulesBelveze) [reply front/components/data_source/DeleteStaticDataSourceDialog.tsx:62]: 😇

**Comment by JulesBelveze** [comment]
Note: we are using `Checkbox` here as our `RadioButton` isn't flexible enough to allow a onClick callback on a disabled item and to render elements in between labels. We are aiming to refactor RadioButton.

We want to refactor our `RadioButton` to use the `RadioGroup` component from [radix-ui](https://www.radix-ui.com/primitives/docs/components/radio-group). Once this is done we can switch back to use it.

cc @flvndvd

**Pull request merged**